### ISSUE TITLE
deprecating/removing k_alert API

### ISSUE THREAD

**Issue by pabigot** [zephyrproject-rtos/zephyr#12232]
tl;dr: `k_alert` is a fairly large API that has extremely limited use and is easily replaced.  Can we eliminate it?

While looking for a communication mechanism for processing notifications I discovered the [alert API](https://docs.zephyrproject.org/latest/kernel/synchronization/alerts.html).  I then looked for places where it's used. Ignoring its own test suite it's used only for the Intel GNA driver (added two months ago) and a latency benchmark, both of which are easily converted to using a workqueue.

From the logs it appears `k_alert` was designed as an event notification structure, but it wasn't used much.  Prior to its use in intel_gna it was last seen in October 2016 when its use in a timer example was converted to a workqueue (56daa2034566b4209c112860f62cb61627338ce3), possibly as steps towards deprecating it.

Locally I updated intel_gna and the benchmark and removed k_alert completely; sanitycheck passes.

APIs that are no longer used should be deprecated and, as soon as feasible, removed, if for no other reason than to prevent new users from being misled into using them instead of more favored solutions.

Is that the right approach in this case?  What's the process for doing this?  If it's going to be deprecated, it should be marked as such in 1.14-LTS lest it be granted immortality due to new out-of-tree uses.

**Comment by galak** [comment]
If we are going to remove this it should be done w/regards to LTS.

**Comment by andrewboie** [comment]
I agree, let's remove it completely.

**Comment by andyross** [comment]
No complaints here.  An API pruning is absolutely a needed thing, we have a lot of oddball stuff floating around.

**Comment by pabigot** [comment]
> @pabigot you have a patch? :-)

The reward for making a suggestion is the opportunity to fix the problem, eh?   Yeah, I made one a couple weeks ago; rebased and checked the updated docs look good.  #12425.
